contract_comm/evm.go

```diff
@@ -124,6 +130,64 @@ func Transfer(db vm.StateDB, sender, recipient common.Address, amount *big.Int)
 	db.AddBalance(recipient, amount)
 }
 
+// GetParentSealFn returns a GetParentSeal function that returns the aggregated parent seal at the given block number.
+// Note: Unless previously cached, retrieves every block between the reference block and n.
```

Comment:
Because retrieving by just the block number is only available on the canonical chain.

Reply:
I think that's okay for now, since doing it this way is an extremely effective DoS vector, since I can cause everyone to iterate over every single header in the chain for cheap. This sort of caching only works in getBlockHash because the limit is 256 blocks, but here our limit is so large that we can't afford to cache.

Reply:
As discussed this morning, this is changed to get get the block by number and I've created #775 to start talking about efforts to make that operation safe and fast.